### PR TITLE
🐛 fix services:access endpoint for product group

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- Issue reported: [#1540](https://github.com/ITISFoundation/osparc-issues/issues/1540) A share permission warning is incorrectly popping up. The reason is that listing user groups filters out Products because Product groups are resources with Read access set to False. In a previous PR, I already prepared the groundwork for this fix by renaming the previous function `list_all_user_groups` to `list_user_groups_with_read_access`. The new function, `list_all_user_groups`, now correctly returns all groups of the user.

## Related issue/s
- closes https://github.com/ITISFoundation/osparc-issues/issues/1540 

## How to test
` pytest tests/unit/with_dbs/02/test_projects_nodes_handlers__services_access.py`

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by elisabettai — APPROVED** [review]
Yeeeeeeeeeeeeah. 🎉 🎉 🎉 
Also see the same case on fogbugz (we should tag it properly when this one goes to staging): https://z43.manuscript.com/f/cases/202180/Wrong-message-when-sharing-a-project-on-sim4life-io-science

**Review by pcrespov — COMMENTED** [review]
Can you pls create a test that reproduces  #1540? 
The test should fail with the previous version and pass with your fix

**Review by sanderegg — APPROVED** [review]
it would be good, if that is an expected behavior, that there is a test for it. Otherwise anyone might inadvertently break it.

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py [hunk 1]
```diff
@@ -565,10 +565,8 @@ async def get_project_services_access_for_gid(
         _user_id = await get_user_id_from_gid(
             app=request.app, primary_gid=query_params.for_gid
         )
-        _, _user_groups, _ = await list_user_groups_with_read_access(
-            app=request.app, user_id=_user_id
-        )
-        groups_to_compare.update({int(item.get("gid", -1)) for item in _user_groups})
+        _user_groups = await list_all_user_groups(app=request.app, user_id=_user_id)
+        groups_to_compare.update({group.gid for group in _user_groups})
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:569]: Test was added :+1:

**Review by pcrespov — APPROVED** [review]

**Comment by odeimaiz** [comment]
Thanks!

**Pull request merged**